languages/java8/template/src/main/java/algorithmia/algorithm/Algorithm.java

```diff
@@ -0,0 +1,26 @@
+package algorithmia;
+import com.algorithmia.algorithmHandler.*;
+import com.algorithmia.*;
+import java.util.HashMap;
+
+public class Algorithm {
+    String Apply(String input, HashMap<String, String> context) throws Exception{
+            if(context.containsKey("local_file")){
+                return "hello ".concat(input) + " your model file is downloaded here ".concat(context.get("local_file"));
+            }
+        return "hello ".concat(input);
+    }
+    HashMap<String, String> DownloadModel() throws Exception{
+            HashMap<String, String> context = new HashMap<>();
+            AlgorithmiaClient client = Algorithmia.client(System.getenv("ALGORITHMIA_API_KEY"));
```

Comment:
Needs API key? But not Enterprise endpoint?

Reply:
we shouldn't need either. there is a bug in the algorithmia-java client that doesn't read the ALGORITHMIA_API_KEY environment variable properly but it should be fixed in the java client not here

Reply:
Yes I agree, I'll make an additional PR in the java client while I'm there to get that fixed.